modules/core/src/main/java/org/apache/ignite/internal/util/HostAndPortRange.java

```diff
@@ -53,60 +55,99 @@ public static HostAndPortRange parse(String addrStr, int dfltPortFrom, int dfltP
 
         String host;
 
+        String portStr;
         int portFrom;
         int portTo;
 
         if (F.isEmpty(addrStr))
             throw createParseError(addrStr, errMsgPrefix, "Address is empty");
 
-        final int colIdx = addrStr.indexOf(':');
-
-        if (colIdx > 0) {
-            String portFromStr;
-            String portToStr;
+        if (addrStr.charAt(0) == '[') { // IPv6 with port(s)
+            int hostEndIdx = addrStr.indexOf(']');
+            if (hostEndIdx == -1) {
+                throw createParseError(addrStr, errMsgPrefix, "IPv6 is incorrect");
+            }
+            host = addrStr.substring(1, hostEndIdx);
+            if (hostEndIdx == addrStr.length() - 1) { // no port specified, using default
+                portFrom = dfltPortFrom;
+                portTo = dfltPortTo;
+            } else { // port specified
```

Comment:
Code style: `else` should start on a new line

Reply:
Fixed